File: backend/src/main/kotlin/com/example/backend/models/Employe.kt
package com.example.backend.models

import jakarta.persistence.*
import org.hibernate.annotations.ColumnDefault

@Entity
@Table(name = "employe")
class Employe {
  @Id
  @Column(name = "id", nullable = false)
  var id: Int? = null

  @Column(name = "identifiant", length = 100)
  var identifiant: String? = null

  @Column(name = "password", length = 50)
  var password: String? = null

  @Column(name = "codebarre_id")
  var codebarreId: String? = null

  @Column(name = "type", length = 100)
  var type: String? = null

  @ManyToOne(fetch = FetchType.LAZY)
  @JoinColumn(name = "user_id", referencedColumnName = "id", nullable = false)
  var user: User? = null

  @Column(name = "etat", length = 20)
  var etat: String? = null

  @Column(name = "faireReductionMax")
  var faireReductionMax: Int? = null
//  @Column(name = "faire_reduction_max")
//  var faireReductionMax: Int? = null

  @ColumnDefault("0")
  @Column(name = "supprimer")
  var  supprimer: Int? = 0
}

File: backend_py/pharmaPython/app/models/employe.py
from typing import Optional

from pydantic import BaseModel
from sqlalchemy import Column, Integer, String, ForeignKey
from sqlalchemy.orm import relationship
from app.core.db import Base
from app.models.user import User


class Employe(Base):
  __tablename__ = "employe"
  __table_args__ = {"extend_existing": True}

  id = Column(Integer, primary_key=True)
  identifiant = Column(String(100))
  password = Column(String(50))
  codebarre_id = Column(String)
  type = Column(String(100))
  user_id = Column(Integer, ForeignKey("user.id"), nullable=False)
  user = relationship("User", lazy="joined")
  etat = Column(String(20))
  faire_reduction_max = Column(Integer)
  supprimer = Column(Integer, default=0)

class EmployeSchema(BaseModel):

  id : Optional[int]
  identifiant : Optional[str]
  password : Optional[str]
  codebarre_id : Optional[str]
  type : Optional[str]
  user_id : Optional[int]
  etat : Optional[str]
  faire_reduction_max : Optional[int]
  supprimer : Optional[int]

  model_config = {"from_attributes": True}

class EmployeIn(BaseModel):

  identifiant : Optional[str]
  password : Optional[str]
  codebarre_id : Optional[str]
  type : Optional[str]
  user_id : Optional[int]
  etat : Optional[str]
  faire_reduction_max : Optional[int]

